[8/n] Update XNNPACK Submodule Version Part 8 Everything Remaining to get it to work

mcr229 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115714

> **__Note:__** XNNPACK Upgrade is too large in the range of **40k** files and **10m** Lines of code, Thus we break the update of the library into multiple parts. All Parts [1 - n] Must be landed together for it to work. ***This also means If there is a revert. Please revert the Entire Stack.***

This change is everything remaining requiring XNNPACK version to work.

@allow-large-files

Differential Revision: [D52099769](https://our.internmc.facebook.com/intern/diff/D52099769/)

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @albanD

---
submodule
(unblock merge to make ShipIt happy)

mcr229 committed:
[8/n] Update XNNPACK Version Part 8 Everything Remaining to get it to work

> **__Note:__** XNNPACK Upgrade is too large in the range of **40k** files and **10m** Lines of code, Thus we break the update of the library into multiple parts. All Parts [1 - n] Must be landed together for it to work. ***This also means If there is a revert. Please revert the Entire Stack.***

This change is everything remaining requiring XNNPACK version to work.

@allow-large-files

Differential Revision: [D52099769](https://our.internmc.facebook.com/intern/diff/D52099769/)

[ghstack-poisoned]

mcr229 committed:
Update on "[8/n] Update XNNPACK Version Part 8 Everything Remaining to get it to work"


> **__Note:__** XNNPACK Upgrade is too large in the range of **40k** files and **10m** Lines of code, Thus we break the update of the library into multiple parts. All Parts [1 - n] Must be landed together for it to work. ***This also means If there is a revert. Please revert the Entire Stack.***

This change is everything remaining requiring XNNPACK version to work.

allow-large-files

Differential Revision: [D52099769](https://our.internmc.facebook.com/intern/diff/D52099769/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

mcr229 committed:
Update on "[8/n] Update XNNPACK Version Part 8 Everything Remaining to get it to work"


> **__Note:__** XNNPACK Upgrade is too large in the range of **40k** files and **10m** Lines of code, Thus we break the update of the library into multiple parts. All Parts [1 - n] Must be landed together for it to work. ***This also means If there is a revert. Please revert the Entire Stack.***

This change is everything remaining requiring XNNPACK version to work.

allow-large-files

Differential Revision: [D52099769](https://our.internmc.facebook.com/intern/diff/D52099769/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

mcr229 committed:
Update on "[8/n] Update XNNPACK Version Part 8 Everything Remaining to get it to work"


> **__Note:__** XNNPACK Upgrade is too large in the range of **40k** files and **10m** Lines of code, Thus we break the update of the library into multiple parts. All Parts [1 - n] Must be landed together for it to work. ***This also means If there is a revert. Please revert the Entire Stack.***

This change is everything remaining requiring XNNPACK version to work.

allow-large-files

Differential Revision: [D52099769](https://our.internmc.facebook.com/intern/diff/D52099769/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

digantdesai reviewed (approved):
lgtm

digantdesai commented on android/pytorch_android/CMakeLists.txt:
```diff
@@ -160,6 +159,7 @@ else()
 
   if(USE_PYTORCH_QNNPACK)
     list(APPEND pytorch_jni_LIBS pytorch_qnnpack)
+    list(APPEND pytorch_jni_LIBS clog)
```
🤞

mcr229 committed:
Update on "[8/n] Update XNNPACK Version Part 8 Everything Remaining to get it to work"


> **__Note:__** XNNPACK Upgrade is too large in the range of **40k** files and **10m** Lines of code, Thus we break the update of the library into multiple parts. All Parts [1 - n] Must be landed together for it to work. ***This also means If there is a revert. Please revert the Entire Stack.***

This change is everything remaining requiring XNNPACK version to work.

allow-large-files

Differential Revision: [D52099769](https://our.internmc.facebook.com/intern/diff/D52099769/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

huydhn commented:
Thank @albanD for raising the concern about adding `skip-pr-sanity-checks` into this PR.  To provide more context, the bulk of changes come from the 3rd-party XNNPACK update, and this is expected.  So, we could add `skip-pr-sanity-checks` to ack this.

malfet commented:
What is the difference between https://github.com/pytorch/pytorch/pull/115587 and this PR? I assume this one is a re-land? If this is the case, please explain in PR description what have changed. Also, re-base against latest `viable/strict` (right now PR base is 100 commits behind trunk), to reduce the risk of land races

mcr229 commented:
This diff is a reland of https://github.com/pytorch/pytorch/pull/115587. Previously there was a failing job in which libclog.a could not be found when building pytorch_android binary on windows. The difference/fix in this PR is how clog is dynamically linked for pytorch_android build. Since XNNPACK has ditched its clog dependency, it will not always be linked, and instead we get the clog dep from qnnpack, only if USE_PYTORCH_QNNPACK is on. We only need clog for QNNPACK, and so the fix then is that we dynamically link clog only if USE_PYTORCH_QNNPACK is on:

```
 if(USE_PYTORCH_QNNPACK)
    list(APPEND pytorch_jni_LIBS pytorch_qnnpack)
    list(APPEND pytorch_jni_LIBS clog)
  endif()
```

mcr229 commented:
@pytorchbot drci

mcr229 committed:
Update on "[8/n] Update XNNPACK Version Part 8 Everything Remaining to get it to work"


> **__Note:__** XNNPACK Upgrade is too large in the range of **40k** files and **10m** Lines of code, Thus we break the update of the library into multiple parts. All Parts [1 - n] Must be landed together for it to work. ***This also means If there is a revert. Please revert the Entire Stack.***

This change is everything remaining requiring XNNPACK version to work.

allow-large-files

Differential Revision: [D52099769](https://our.internmc.facebook.com/intern/diff/D52099769/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 albanD

[ghstack-poisoned]

osalpekar commented:
@pytorchbot merge -f "already landed internally"
